test/integration_test/applicationbackup_test.go

```diff
@@ -629,3 +724,168 @@ func commonApplicationBackupScheduleTests(
+	require.NoError(t, err, "Error creating app backups")
+
+	// Create backup location on second cluster
+	err = dumpRemoteKubeConfig(remoteConfig)
+	if err != nil {
+		require.NoErrorf(t, err, "Unable to write clusterconfig: %v", err)
+	}
+
+	err = setRemoteConfig(remoteFilePath)
+	require.NoError(t, err, "Error setting remote config")
+
+	// Create namespace for the backuplocation on second cluster
+	ns, err := k8s.Instance().CreateNamespace(appCtx.GetID(),
+		map[string]string{
+			"creator": "stork-test",
+			"app":     appCtx.App.Key,
+		})
+	require.NoError(t, err, "Failed to create namespace %s: %v", appCtx.GetID(), err)
+
+	backupLocation2, err := createBackupLocation(t, backupLocationName, ns.Name, storkv1.BackupLocationS3, "secret-config")
+	require.NoError(t, err, "Error creating backuplocation on second cluster")
+	logrus.Infof("Created application backup on second cluster %s: sync:%t", backupLocation.Name, backupLocation.Location.Sync)
+
+	// Set sync to true on second cluster so that backup location gets synced
+	backupLocation2.Location.Sync = true
+	_, err = k8s.Instance().UpdateBackupLocation(backupLocation2)
+	require.NoError(t, err, "Failed to set backup-location sync to true")
+	logrus.Infof("Updated application backup on 2nd cluster %s: sync:%t", backupLocation2.Name, backupLocation2.Location.Sync)
+
+	// Check periodically to see if the backup from this test is synced on second cluster
+	var allAppBackups *storkv1.ApplicationBackupList
+	listBackupsTask := func() (interface{}, bool, error) {
+		allAppBackups, err = k8s.Instance().ListApplicationBackups(ns.Name)
+		if err != nil || len(allAppBackups.Items) == 0 {
+			logrus.Infof("Failed to list app backups on second cluster. Error: %v. Number of backups: %v", err, len(allAppBackups.Items))
+
+			return "", true, fmt.Errorf("Failed to list app backups on second cluster.")
+		} else if len(allAppBackups.Items) > 0 {
+			// backups sync has started, check if current backup has synced
+			backupToRestore := getBackupFromListWithAnnotations(allAppBackups, firstBackup.Annotations)
+			if backupToRestore != nil {
+				return "", false, nil
+			}
+		}
+		return "", true, fmt.Errorf("Failed to list app backups on second cluster.")
+	}
+	_, err = task.DoRetryWithTimeout(listBackupsTask, defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error listing application backups")
+
+	backupToRestore := getBackupFromListWithAnnotations(allAppBackups, firstBackup.Annotations)
+	require.NotNil(t, backupToRestore, "Backup sync failed. Backup not found on the second cluster")
+
+	// Create application restore using the backup selected, on second cluster
+	logrus.Infof("Starting Restore on second cluster.")
+	appRestoreForBackup, err := createApplicationRestore(t, "mysql-restore-backup-sync", ns.Name, backupToRestore, backupLocation2)
+	require.NotNil(t, appRestoreForBackup, "Backup sync failed. Backup not found on the second cluster")
```

Comment:
Message should be updated to say Restore?

Reply:
Modified message.